brother_ql/backends/helpers.py

```diff
@@ -131,21 +130,142 @@ def status(
+    command = b"\x1b\x69\x55" + setting.to_bytes(1) + b"\x01"
+    if payload is not None:
+        command += payload
+    printer.write(command)
+    result = get_status(printer, receive_only=True, target_status=0xF0)
     return result
+
+
+def write_setting(
+    printer,
+    setting,
+    payload,
+):
+    """
+    Write setting to printer.
+
+    :param BrotherQLBackendGeneric printer: A printer instance.
+    :param int setting: The code for the setting.
+    :param bytes payload: Payload for the setting.
+    """
+
+    # switch to raster command mode
+    printer.write(b"\x1b\x69\x61\x01")
+    # write settings
+    command = b"\x1b\x69\x55" + setting.to_bytes(1) + b"\x00"
+    command += payload
+    printer.write(command)
+    # retrieve status to make sure no errors occured
+    result = get_status(printer)
+    if result['status_code'] != 0x0:
+        raise ValueError("Failed to modify settings")
+    return result
+
+
+def configure(
+    printer_identifier=None,
+    backend_identifier=None,
+    action="get",
+    key=None,
+    value=None,
+):
+    """
+    Read or modify power settings.
+
+    :param str printer_identifier: Identifier for the printer
+    :param str backend_identifier: Can enforce the use of a specific backend
+    :param str action: Action to perform, get or set
+    :param str key: Key name for the settings
+    :param int value: Value to update for the specified key
+    """
+
+    printer=get_printer(printer_identifier, backend_identifier)
+
+    if action not in ['get', 'set']:
+        raise ValueError(f"Invalid action '{action}'")
+    if key not in ['power-off-delay', 'auto-power-on']:
+        raise ValueError(f"Invalid key '{key}'")
+    if action == 'set':
+        if value is None:
+            raise ValueError(f"Specify a valid value for key '{key}'")
+
+    series_code = get_status(printer, target_status=0x0)['series_code']
+
+    if action == 'set':
+        if key == 'auto-power-on':
+            payload = value.to_bytes(1)
+            write_setting(printer, 0x70, payload)
+            get_status(printer, 0x0)
+        elif key == 'power-off-delay':
+            payload = b''
+            # 0x30 series needs an extra byte here
+            if series_code == 0x30:
+                payload += b"\x00"
+            payload += value.to_bytes(1)
+            write_setting(printer, 0x41, payload)
+            get_status(printer, 0x0)
+        else:
+            raise ValueError(f"Key {key} is invalid")
+
+    # retrieve settings
+    retrieved_val = None
+    if key == 'auto-power-on':
+        retrieved_val = get_setting(printer, 0x70)['setting']
+    elif key == 'power-off-delay':
+        payload = b''
+        # 0x30 series needs an extra byte here
+        if series_code == 0x30:
+            payload += b"\x00"
+        retrieved_val = get_setting(printer, 0x41, payload)['setting']
+    else:
+        raise ValueError(f"Key {key} is invalid")
+
+    print(f"{key}: {retrieved_val}")
```

Comment:
C (non blocking): I think it would make sense to either `logger.info` this or return it, print is pretty hard to work with reliably.

Reply:
I'm using print because these should always be printed regardless of log levels.

Reply:
Oh, there's no way to get the value if you're calling this function, now I see what you mean. I'll return the value.

Reply:
Sorry if that was not clear; english is a second language for me so please ask if my thoughts are not articulated unclearly